src/functional/iterator/transforms/popup_tmps.py

```diff
@@ -77,6 +80,9 @@ def wrap(fun: ir.Lambda, args: list[ir.Expr]) -> ir.FunCall:
                         fun=ir.SymRef(id="scan"), args=scan_args
                     )
                 elif is_reduce:
+                    assert isinstance(node.fun, ir.FunCall) and isinstance(
+                        node.fun.args[0], ir.FunCall
+                    )
```

Comment:
These checks should validate `node` has the proper structure, but this one appears to only include stuff that is needed to make mypy happy. For example the length of the arguments is not validated, but probably should be. I can not comment on the actual check because I am missing the knowledge of the actual pass here.

Reply:
it can be treated as other asserts, i.e. I can add the comment "# just for mypy" or analogous